src/gui.js

```diff
@@ -5583,17 +5583,84 @@ IDE_Morph.prototype.logout = function () {
     );
 };
 
+IDE_Morph.prototype.buildProjectRequest = function () {
+    var xml = this.serializer.serialize(this.stage),
+        thumbnail = normalizeCanvas(
+            this.stage.thumbnail(
+                SnapSerializer.prototype.thumbnailSize
+        )).toDataURL(),
+        body;
+
+    this.serializer.isCollectingMedia = true;
+    body = {
+        notes: this.projectNotes,
+        xml: xml,
+        media: this.hasChangedMedia ?
+            this.serializer.mediaXML(this.projectName) : null,
+        thumbnail: thumbnail,
+        remixID: this.stage.remixID
+    };
+    this.serializer.isCollectingMedia = false;
+    this.serializer.flushMedia();
+
+    return body;
+}
+
+IDE_Morph.prototype.verifyProject = function (body) {
+    // Ensure the project is less than 10MB and serializes correctly.
+    var encodedBody = JSON.stringify(body);
```

Comment:
You'll note that the 10MB limit is on the entire body not just the project + media. Mostly this means we lose ~.4KB to the thumbnail + JSON compared to before, but up to ~1MB based on logs I saw. We could possibly raise the cloud server limit by a bit to loosen things.